tests/integration/kubernetes/gha-run.sh

```diff
@@ -236,6 +239,76 @@ function cleanup() {
+		*) >&2 echo "${SNAPSHOTTER} flavour is not supported"; exit 2 ;;
+	esac
+
+	echo "::endgroup::"
+}
+
+function deploy_nydus_snapshotter() {
+    echo "deploy_nydus_snapshotter"
+	NODE_NAME=${NODE_NAME:-$(hostname | tr A-Z a-z)} 
+    kubectl config set-context --current --namespace=default
+	nydus_snapshotter_version=$(get_from_kata_deps "externals.nydus-snapshotter.version")
+    sed -i -e "s|ghcr.io/containerd/nydus-snapshotter:latest|ghcr.io/containerd/nydus-snapshotter:${nydus_snapshotter_version}|g" "${NYDUS_SNAPSHOTTER_DEPLOY_DIR}/nydus-snapshotter.yaml"
+
+    sed -i -e "s|CONTAINERD_CONFI_DIR|${CONTAINERD_CONFI_DIR}|g" "${tools_dir}/tools/nydus-snapshotter.yaml"
+    kubectl apply -f "${NYDUS_SNAPSHOTTER_DEPLOY_DIR}/nydus-snapshotter-rbac.yaml"
+    kubectl apply -f "${NYDUS_SNAPSHOTTER_DEPLOY_DIR}/nydus-snapshotter.yaml"
+	kubectl label node "$NODE_NAME" --overwrite nydus-snapshotter/nydus-snapshotter=true
+}
+
+function cleanup_snapshotter() {
+	echo "::group::Cleanuping ${SNAPSHOTTER}"
+
+	case ${SNAPSHOTTER} in
+		nydus) cleanup_nydus_snapshotter ;;
+		*) >&2 echo "${SNAPSHOTTER} flavour is not supported"; exit 2 ;;
+	esac
+
+	echo "::endgroup::"
+}
+
+function cleanup_nydus_snapshotter() {
+    echo "cleanup_nydus_snapshotter"
+	NODE_NAME=${NODE_NAME:-$(hostname | tr A-Z a-z)} 
+    if [ ! -z "$(pidof containerd-nydus-grpc)" ]; then
+        for i in $(ctr -n k8s.io snapshot --snapshotter nydus list | grep -v KEY | cut -d' ' -f1); do
+            ctr -n k8s.io snapshot --snapshotter nydus rm $i || true
+        done
+        echo "Removing nydus snapshotter from containerd"
+        containerd_imports_path="${CONTAINERD_CONFI_DIR}/config.toml.d"
+        sed -i -e "s|\"${containerd_imports_path}/nydus-snapshotter.toml\"||g" ${CONTAINERD_CONFI_DIR}/config.toml
+        sed -i -e "s|, ]|]|g" ${CONTAINERD_CONFI_DIR}/config.toml
+
+        if grep -q "${snapshot_annotations_marker}" "${CONTAINERD_CONFI_DIR}/config.toml"; then
+            sed -i '/'"${snapshot_annotations_marker}"'/d' \
+                "${CONTAINERD_CONFI_DIR}/config.toml"
+            sed -i '/disable_snapshot_annotations = false/d' \
+                "${CONTAINERD_CONFI_DIR}/config.toml"
+        else
+            sed -i -e "s|disable_snapshot_annotations = false|disable_snapshot_annotations = true|" ${CONTAINERD_CONFI_DIR}/config.toml
+        fi
+    fi
+
+    kubectl delete -f "${NYDUS_SNAPSHOTTER_DEPLOY_DIR}/nydus-snapshotter.yaml"
+    kubectl delete -f "${NYDUS_SNAPSHOTTER_DEPLOY_DIR}/nydus-snapshotter-rbac.yaml"
+    sudo systemctl restart containerd
+    sudo -E kill -9 $(pidof containerd-nydus-grpc) || true
+    echo "Removing nydus-snapshotter artifacts from host"
+    sudo rm -f /usr/local/bin/containerd-nydus-grpc
+    sudo rm -f /usr/local/bin/nydus-overlayfs
+    sudo rm -rf /etc/nydus
+    sudo rm -rf /var/lib/containerd-nydus/*
+
+	kubectl label node "$NODE_NAME" --overwrite nydus-snapshotter/nydus-snapshotter=false
```

Comment:
ca we add the label to worker nodes as https://github.com/confidential-containers/cloud-api-adaptor/blob/main/libvirt/install_operator.sh#L22
```
	local workers
	local label="nydus-snapshotter/nydus-snapshotter=false"

	workers="$(kubectl get nodes --no-headers | grep '\<worker\>' | awk '{ print $1 }')"
	for nodename in $workers; do
		if ! kubectl get "node/${nodename}" -o jsonpath='{.metadata.labels}' | \
			grep "$label" >/dev/null; then
			kubectl label node "$nodename" --overwrite "${label}"
		fi
	done
```

Reply:
Done.